### DISCUSSION TITLE
Possibility to specify name for .bin files emmited from partition command?

### DISCUSSION THREAD

**Discussion by oddnugget** [donmccurdy/glTF-Transform#412] [Q&A] [answered]
Currently the partition command just names the `.bin` files `mesh.bin`. I notice in the code that it defaults to that if mesh.getName returns undefined. My question is how can i make sure the name of the mesh is persisted? The source gltf comes from threejs gltf export, and there the geometry does have a name. 

Alternatively, is there a way to specify what i would like to name the bin file?

**Comment by donmccurdy** [comment, marked as answer]
If `mesh.getName()` is returning undefined, it's likely that GLTFExporter has not saved the name of the geometry in a location that's useful to the `partitiion()` function. glTF has "Mesh" and "Primitive" concepts, but neither is really a 1:1 counterpart to THREE.BufferGeometry.

You may need to use the scripting API to have more control of the bin file name, something like:

```js
import { NodeIO } from '@gltf-transform/core';
import { KHRONOS_EXTENSIONS } from '@gltf-transform/extensions';
import { partition } from '@gltf-transform/functions';

const io = new NodeIO().registerExtensions( KHRONOS_EXTENSIONS );

const document = io.read('./input.glb');

// Partition.
await document.transform( partition( { meshes: true } ) );

// Rename buffer(s).
let index = 0;
for (const buffer of document.getRoot().listBuffers()) {
  buffer.setURI(`my_buffer_${index++}.bin`);
}

io.write('./output.glb', document);
```

**Reply by oddnugget** [reply]
Thanks for the answer! I ended up doing something similar as well, though in my case there is only one mesh per gltf so i can safely set the name using `buffer.setUri('name.bin')`
